akka-http-proxy/src/main/scala/com/pagerduty/akka/http/proxy/HttpProxy.scala

```diff
@@ -28,18 +26,12 @@ class HttpProxy[AddressingConfig](
       upstream.addressRequestWithOverrides(request, addressingConfig)
 
     val proxyRequest =
-      targetedRequest
-        .removeHeader(TimeoutAccessHeaderName) // Akka HTTP server adds the Timeout-Access for internal reasons, but it should not be proxied
-        .removeHeader(Connection.name) // remove any existing Connection header
-        .addHeader(Connection(KeepAliveHeaderValue)) // add a Connection: keep-alive header
-
-    val modifiedProxyRequest = proxyRequestModifier match {
-      case Some(modifier) => modifier(proxyRequest)
-      case None => proxyRequest
-    }
+      targetedRequest.removeHeader(TimeoutAccessHeaderName) // Akka HTTP server adds the Timeout-Access for internal reasons, but it should not be proxied
```

Comment:
Is it not possible to chain this `.removeHeader` to `upstream.addressRequestWithOverrides()`, since we don't really need `targetedRequest` by itself?

Reply:
True, it could be cleaned up. Fixed.

Reply:
👍